### ISSUE TITLE
Raspberry Pi 3 compilation error

### ISSUE THREAD

**Issue by artus9033** [lovell/shorter#1]
After running the command npm install --save shorter, I get the following error:
`pi@raspberrypi:~/Desktop/rpi $ npm install --save shorter

> shorter@0.1.0 install /home/pi/Desktop/rpi/node_modules/shorter
> node-gyp rebuild

make: Wejście do katalogu '/home/pi/Desktop/rpi/node_modules/shorter/build'
  CC(target) Release/obj.target/shoco/src/shoco/shoco.o
*** Error in `cc': double free or corruption (top): 0x01d2e1e0 ***
src/shoco.target.mk:99: polecenia dla obiektu 'Release/obj.target/shoco/src/shoco/shoco.o' nie powiodły się
make: *** [Release/obj.target/shoco/src/shoco/shoco.o]: Przerwane
make: Opuszczenie katalogu '/home/pi/Desktop/rpi/node_modules/shorter/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/pi/n/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:285:23)
gyp ERR! stack     at emitTwo (events.js:125:13)
gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:197:12)
gyp ERR! System Linux 4.9.24-v7+
gyp ERR! command "/home/pi/n/bin/node" "/home/pi/n/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/pi/Desktop/rpi/node_modules/shorter
gyp ERR! node -v v8.0.0
gyp ERR! node-gyp -v v3.6.1
gyp ERR! not ok 
npm WARN RaspCopter@1.0.0 No repository field.
npm WARN RaspCopter@1.0.0 No license field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! shorter@0.1.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the shorter@0.1.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2017-07-10T15_42_30_811Z-debug.log
`
Full npm log:
[2017-07-10T15_42_30_811Z-debug.log.txt](https://github.com/lovell/shorter/files/1136235/2017-07-10T15_42_30_811Z-debug.log.txt)

NPM version: 5.0.0
Node version: 8.0.0

What's interesting, installing shorter v0.0.1 was succesful. Please fix this problem.

**Comment by lovell** [comment]
Hello, you've run into https://gcc.gnu.org/bugzilla/show_bug.cgi?id=70132 triggered by https://github.com/lovell/shorter/blob/master/binding.gyp#L16

I'll need to update `binding.gyp` to avoid this flag on ARM.

In the meantime, the following workaround should help:
```sh
CFLAGS="-march=armv7-a" CXXFLAGS="-march=armv7-a" npm install shorter
```

**Comment by artus9033** [comment]
Right, that helped, thanks. But I have another problem: I'm building an app that communicates from a pc with the rpi and the app on the pc side utilizes electron and the problem is that compilation throws no errors, but after I run the app and require shorter, I get the following:
`[nodemon] starting `electron main.js`

App threw an error during load
Error: Procedura inicjowania biblioteki do┬│┬╣czanej dynamicznie (DLL) nie powiod┬│a si├¬.
\\?\C:\Users\Artur\Documents\Projekty Aplikacji\Projekty Roboty\RaspCopter\source\RaspController\electron\node_modules\shorter\build\Release\shorter.node
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:173:20)
    at Object.Module._extensions..node (module.js:598:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:173:20)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\Users\Artur\Documents\Projekty Aplikacji\Projekty Roboty\RaspCopter\source\RaspController\electron\node_modules\shorter\index.js:3:17)
    at Object.<anonymous> (C:\Users\Artur\Documents\Projekty Aplikacji\Projekty Roboty\RaspCopter\source\RaspController\electron\node_modules\shorter\index.js:27:3)`
I suspect electron for this error. Is there any fix for that?

**Comment by artus9033** [comment]
Alright, I finally got it working by cloning the shorter repo, changing the architecture to armv7-a in shorter's binding.gyp & in shoco's binding.gyp and then installing the module from file. Also, I got it working with electron by using the electron-rebuild module to rebuild all native packages (of course including shorter) for electron.

**Comment by lovell** [comment]
Commit b885b1f ensures `-march=native` is no longer set on ARM machines and this is now available in v0.2.0. Thanks for reporting!

**Comment by artus9033** [comment]
Glad to help. Thank you too!
